src/Modules/UngradedModules/FreeMod.jl

```diff
@@ -308,3 +308,67 @@ end
 
 rels(F::FreeMod) = elem_type(F)[]
 
+function syzygy_generators(
+    g::Vector{T};
+    parent::Union{<:FreeMod, Nothing}=nothing
+  ) where {T<:FreeModElem}
+  isempty(g) && return Vector{T}()
+  F = Oscar.parent(first(g))
+  @assert all(Oscar.parent(x) === F for x in g) "parent mismatch"
+  R = base_ring(F)
+  m = length(g)
+  G = (parent === nothing ?  FreeMod(R, m) : parent)::typeof(F)
+  @assert ngens(G) == m "given parent does not have the correct number of generators"
+  phi = hom(G, F, g)
+  K, _ = kernel(phi)
+  return ambient_representatives_generators(K)
+end
+
+@doc raw"""
+    syzygy_generators(
+        a::Vector{T};
+        parent::Union{FreeMod{T}, Nothing} = nothing
+      ) where {T<:RingElem}
+
+Return generators for the syzygies on the polynomials given as elements of `a`.
+The optional keyword argument can be used to specify the parent of the output.
+
+# Examples
+```jldoctest
+julia> R, (x, y) = polynomial_ring(QQ, [:x, :y])
+(Multivariate polynomial ring in 2 variables over QQ, QQMPolyRingElem[x, y])
+
+julia> S = syzygy_generators([x^3+y+2,x*y^2-13*x^2,y-14])
+3-element Vector{FreeModElem{QQMPolyRingElem}}:
+ (-y + 14)*e[2] + (-13*x^2 + x*y^2)*e[3]
+ (-169*y + 2366)*e[1] + (-13*x*y + 182*x - 196*y + 2744)*e[2] + (13*x^2*y^2 - 2548*x^2 + 196*x*y^2 + 169*y + 338)*e[3]
+ (-13*x^2 + 196*x)*e[1] + (-x^3 - 16)*e[2] + (x^4*y + 14*x^4 + 13*x^2 + 16*x*y + 28*x)*e[3]
+```
+"""
+function syzygy_generators(
+    a::Vector{T};
+    parent::Union{FreeMod{T}, Nothing} = nothing
+  ) where {T<:RingElem}
+  isempty(a) && return Vector{FreeModElem{T}}()
+  R = Oscar.parent(first(a))
+  @assert all(Oscar.parent(x) === R for x in a) "parent mismatch"
+  F = FreeMod(R, 1)
+  return syzygy_generators(elem_type(F)[x*F[1] for x in a]; parent)
+end
+
+function syzygy_generators(
+    a::Vector{T};
+    parent::Union{FreeMod{T}, Nothing} = nothing
+  ) where {CT <: Union{<:FieldElem, ZZRingElem, QQPolyRingElem}, # Can be adjusted to whatever is digested by Singular
```

Comment:
@wdecker , @joschmitt : Your booktests in `H3.jlcon` were failing because I restricted the signature of the original method for `syzygy_generators`. In your example there the syzygies are computed in a ring which is of type `MPolyDecRing{QQPolyRingElem, AbstractAlgebra.Generic.MPolyRing{QQPolyRingElem}}`, so we have a tower of a multivariate, graded polynomial ring over a univariate polynomial ring. I was unaware that Singular can apparently handle this coefficient ring do reasonable computations over it. 

Could you please double-check that this is really what you wanted and that this is OK? I.e. to also allow coefficients of type `QQPolyRingElem` here? Thanks!

Reply:
> I.e. to also allow coefficients of type `QQPolyRingElem` here?

It worked correctly in the example, so I did not double check. The old code is here:

function syzygy_generators(a::Vector{<:MPolyRingElem})
  I = ideal(a)
  s = Singular.syz(singular_generators(I))
  F = free_module(parent(a[1]), length(a))
  @assert rank(s) == length(a)
  return [F(s[i]) for i=1:Singular.ngens(s)]
end

@ederc, @hannes14 What is happening here in singular.jl/Singular?

Reply:
P.S. If Singular is working with the fraction field of T, I could rewrite the example accordingly

Reply:
Then I suppose that it's OK as is now. Everything is executed with the old code again.